runway/blueprints/staticsite/auth_at_edge.py

```diff
@@ -0,0 +1,434 @@
+            lambda_execution_role
+        )
+
+        # CloudFront Distribution
+        distribution_options = self.get_distribution_options(
+            bucket,
+            oai,
+            check_auth_lambda['version'],
+            http_headers_lambda['version'],
+            parse_auth_lambda['version'],
+            refresh_auth_lambda['version'],
+            sign_out_lambda['version']
+        )
+        distribution = self.add_cloudfront_distribution( # noqa pylint: disable=unused-variable
+            allow_access,
+            distribution_options
+        )
+
+    def add_lambda(self,
+                   title,  # type: str
+                   description,  # type: str
+                   handle,  # type: str
+                   role  # iam.Role
+                   ):
+        # type: (...) -> Dict[str, Any]
+        """Creates a lambda function and it's version.  # noqa
+
+        Args:
+            title (str): The name of the function in PascalCase
+            description (str): Description to be displayed in the
+                lambda panel
+            handle (str): The underscore separated representation
+                of the name of the lambda. This handle is used to
+                determine the handler for the lambda as well as
+                identify the correct Code hook_data information.
+            role (IAM.Role): The Lambda Execution Role
+        """
+        function = self.get_auth_at_edge_lambda(
+            title,
+            description,
+            handle,
+            role
+        )
+
+        return {
+            'function': function,
+            'version': self.add_version(title, function)
+        }
+
+    def get_auth_at_edge_lambda(self,
+                                title,   # type: str
+                                description,   # type: str
+                                handler,  # type: str
+                                role  # type: iam.Role
+                                ):
+        # type: (...) -> awslambda.Function
+        """Create an Auth@Edge lambda resource.
+
+        Args:
+            name (str): The name of the function in PascalCase
+            description (str): Description to be displayed in the
+                lambda panel
+            handle (str): The underscore separated representation
+                of the name of the lambda. This handle is used to
+                determine the handler for the lambda as well as
+                identify the correct Code hook_data information.
+            role (IAM.Role): The Lambda Execution Role
+        """
+        return self.template.add_resource(
+            awslambda.Function(
+                title,
+                Code=self.context.hook_data['aae_lambda_config'].get(handler),
+                Description=description,
+                Handler='__init__.handler',
+                Role=role.get_att("Arn"),
+                Runtime='python3.7'
+            )
+        )
+
+    def add_version(self,
```

Comment:
For the methods that return something specific and ment to be consumed by something specific, it may be beneficial to document them as such for maintainability.

example
```
        """Create a version association with a Lambda@EDGE function.
        In order to ensure different versions of the function
        are appropriately uploaded a hash based on the code of the
        lambda is appended to the name. As the code changes so
        will this hash value.
        Args:
            title (str): The name of the function in PascalCase
            role (awslambda.Function): The Lambda function

        Returns:
            Calculated MD5 hash to define a Lambda function version.
            This version is needed for CloudFront distribution configuration.

        """

```